modules/tinymce/src/core/test/ts/webdriver/PageUpDownKeysTest.ts

```diff
@@ -0,0 +1,102 @@
+      TinySelections.setCursor(editor, [ 0, 1 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+      TinyContentActions.keystroke(editor, Keys.pageUp());
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 0);
+    });
+
+    it('TINY-4612: macOS wont move the caret if it is not at inline element at the start of the line, whereas other OSs do', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link2</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 2 ], 0);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+      TinyContentActions.keystroke(editor, Keys.pageUp());
+
+      if (platform.os.isOSX() || platform.browser.isFirefox()) {
+        TinyAssertions.assertCursor(editor, [ 0, 2 ], 0);
+      } else {
+        TinyAssertions.assertCursor(editor, [ 0, 0 ], 0);
+      }
+    });
+
+    it('TINY-4612: inline_boundaries: false macOS doesnt move, whereas other OSs do', async () => {
+      const editor = hook.editor();
+      editor.settings.inline_boundaries = false;
+      editor.setContent('<p>test<a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 2 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+      TinyContentActions.keystroke(editor, Keys.pageUp());
+
+      if (platform.os.isOSX() || platform.browser.isFirefox()) {
+        TinyAssertions.assertCursor(editor, [ 0, 2 ], 2);
+      } else {
+        TinyAssertions.assertCursor(editor, [ 0, 0 ], 0);
+      }
+      delete editor.settings.inline_boundaries;
+    });
+  });
+
+  context('Page Down', () => {
+    it('TINY-4612: caret should be placed at the end of the line if the last child is an inline element', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link</a></p>');
+      TinySelections.setCursor(editor, [ 0, 0 ], 0);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+      TinyContentActions.keystroke(editor, Keys.pageDown());
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 2);
+    });
+
+    it('TINY-4612: macOS wont move the caret if it is not at inline element at the end of the line, whereas other OSs do', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p>test<a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 0 ], 0);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+      TinyContentActions.keystroke(editor, Keys.pageDown());
+
+      if (platform.os.isOSX() || platform.browser.isFirefox()) {
+        TinyAssertions.assertCursor(editor, [ 0, 0 ], 0);
+      } else {
+        TinyAssertions.assertCursor(editor, [ 0, 2 ], 4);
+      }
+    });
+
+    it('TINY-4612: inline_boundaries: false macOS doesnt move, whereas other OSs do', async () => {
```

Comment:
```suggestion
    it('TINY-4612: "inline_boundaries: false" macOS doesnt move, whereas other OSs do', async () => {
```

Reply:
All assertions related to "inline boundaries: false" were removed because Windows and Linux browsers always move the caret with PageUp/PageDown, and on Mac, these tests are skipped.

Reply:
These shouldn't have been removed, as they assert the selection isn't moved to the incorrect location by the new logic.

Reply:
updated